### PR TITLE
connection type preview form generation

### PR DESCRIPTION
**Pull request by christianvogt**
https://issues.redhat.com/browse/RHOAIENG-10319

## Description

Generates a preview form from the connection type resource and field descriptors.
Uses the existing mock which encompasses all variations of fields.

Form fields are rendered in their default states but do not react to changes so they act as read only.

Provides a drawer component to render the content preview given the connection type resource.

There are still several UX discussions on going but any changes should be minimal to incorporate.
There's also a quirk with dropdowns that is causing the drawer to scroll back to the top whenever it is opened. This can be dealt with later.

image
image
image
image
image
image
image
image
image
image

## How Has This Been Tested?

Manually. Test pages.
Added unit tests.

## Test Impact

Unit tests.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

cc @simrandhaliw

### CODE REVIEW

**Comment by christianvogt** [comment]
@emilys314 @jeff-phillips-18 rebased this PR

**Comment by emilys314** [comment]
Looks good to me I think

File: frontend/src/concepts/connectionTypes/fields/DropdownFormField.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import DataFormFieldGroup from '~/concepts/connectionTypes/fields/DataFormFieldGroup';
+
+type Props = {
+  field: DropdownField;
+  isPreview?: boolean;
+  onChange?: (value: string[]) => void;
+  value?: string[];
+};
+
+const DropdownFormField: React.FC<Props> = ({ field, isPreview, onChange, value }) => {
```
Comment (jeff-phillips-18) [inline frontend/src/concepts/connectionTypes/fields/DropdownFormField.tsx:13]: Nit: Calling it a Dropdown but using a Select could be confusing.

Comment (christianvogt) [reply frontend/src/concepts/connectionTypes/fields/DropdownFormField.tsx:13]: That's fair. It coincides with the type. And the type name will be shown to the user as an available type.

Perhaps we should discuss the type name with us?

**Pull request merged**